### PR TITLE
adds Group resource

### PR DESCRIPTION
**Pull request by jaypipes**
Adds resource management for Group resources in the IAM API.

Signed-off-by: Jay Pipes <jaypipes@gmail.com>

By submitting this pull request, I confirm that my contribution is made
under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/resource/group/hooks.go [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+		ctx, input, func(page *svcsdk.ListAttachedGroupPoliciesOutput, _ bool) bool {
+			if page == nil {
+				return true
+			}
+			for _, p := range page.AttachedPolicies {
+				res = append(res, p.PolicyArn)
+			}
+			return page.IsTruncated != nil && *page.IsTruncated
+		},
+	)
```
Comment (RedbackThomson) [inline pkg/resource/group/hooks.go:95]: Cool

File: test/e2e/common/types.py [hunk 2]
```diff
@@ -1,2 +1,3 @@
+GROUP_RESOURCE_PLURAL = 'groups'
 ROLE_RESOURCE_PLURAL = 'roles'
 POLICY_RESOURCE_PLURAL = 'policies'
```
Comment (RedbackThomson) [inline test/e2e/common/types.py:3]: I don't really care for this file. I think it'd be better to just have these defined as consts in each of the test files, then import them as necessary into other test files

**Review by RedbackThomson — APPROVED** [review]
Just a small nitpick, otherwise looks good

File: test/e2e/tests/test_group.py [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+        time.sleep(CHECK_STATUS_WAIT_SECONDS)
+
+        # Before we update the Group CR below, let's check to see that the
+        # MaxSessionDuration field in the CR is still what we set in the
+        # original Create call.
+        cr = k8s.get_resource(ref)
+        assert cr is not None
+        assert 'spec' in cr
+        assert 'name' in cr['spec']
+        assert cr['spec']['name'] == group_name
```
Comment (A-Hilaly) [inline test/e2e/tests/test_group.py:68]: This is not needed for `Group` resource. `MaxSessionDuration` is not part of the spec here

**Review by A-Hilaly — COMMENTED** [review]
LGTM overall, leaving one comment

File: config/iam/recommended-inline-policy [hunk 4]
```diff
@@ -17,6 +21,7 @@
                 "iam:DeletePolicyVersion",
                 "iam:ListPolicyVersions",
                 "iam:ListPolicyTags",
+                "iam:ListAttachedGroupPolicies",
```
[the remark this answers is not in the document]
Comment (jaypipes) [reply config/iam/recommended-inline-policy:24]: nice catch. yes there is :)

File: test/e2e/tests/test_group.py (hunk 3 above)
Comment (jaypipes) [reply test/e2e/tests/test_group.py:68]: will remove.

**Comment by jaypipes** [comment]
@A-Hilaly @RedbackThomson OK, updated. thanks for the reviews!

**Review by A-Hilaly — APPROVED** [review]
Nice!
/lgtm

**Pull request merged**